test/integration/network_reserved_ips_test.go

```diff
@@ -2,150 +2,199 @@ package integration
-	if getReservedIpError != nil {
-		t.Logf("Error getting ipaddress, expected struct, got %v and error %v", ipaddress, getReservedIpError)
+	ip, getErr := client.GetReservedIPAddress(context.Background(), address)
+	if getErr == nil {
+		t.Errorf("Expected error when getting IP address due to insufficient permissions, but got none")
+	} else {
+		t.Logf("Correctly received %v for IP Address and error when getting IP address: %v", ip, getErr)
 	}
 
 	// Attempt to delete a reserved IP address
-	addressToBeDeleted := "172.28.3.4"
-	deleterr := client.DeleteReservedIPAddress(context.Background(), addressToBeDeleted)
-	if deleterr != nil {
-		t.Logf("Error deleting reserved IP address: %v", deleterr)
+	delAddr := "172.28.3.4"
+	delErr := client.DeleteReservedIPAddress(context.Background(), delAddr)
+	if delErr == nil {
+		t.Errorf("Expected error when deleting IP address due to insufficient permissions, but got none")
 	} else {
-		t.Logf("Successfully deleted reserved IP address: %s", addressToBeDeleted)
+		t.Logf("Correctly received error when deleting IP address: %v", delErr)
 	}
 }
 
 // TestReservedIPAddresses_EndToEndTest performs an end-to-end test of the Reserved IP functionality
 // for users with the can_reserve_ip flag enabled
 func TestReservedIPAddresses_EndToEndTest(t *testing.T) {
-	client, teardown := createTestClient(t, "fixtures/TestReservedIpAddresses_EndToEndTest")
+	client, teardown := createTestClient(t, "fixtures/TestReservedIPAddresses_EndToEndTest")
 	defer teardown()
 
 	filter := ""
-	ipList, listIpErr := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
-	if listIpErr != nil {
-		t.Logf("Error listing ipaddresses, expected struct, got error %v", listIpErr)
-	} else {
-		if len(ipList) == 0 {
-			t.Logf("The customer has not reserved an IP %v", ipList)
-		}
+
+	ipList, err := client.ListReservedIPAddresses(context.Background(), NewListOptions(0, filter))
+
+	if err != nil {
+		t.Fatalf("Error listing IP addresses: %v", err)
 	}
 
+	initialCount := len(ipList)
+
 	// Attempt to reserve an IP
-	reserveIP, reserveIpErr := client.ReserveIPAddress(context.Background(), ReserveIPOptions{
+	resIP, resErr := client.ReserveIPAddress(context.Background(), ReserveIPOptions{
 		Region: "us-east",
 	})
-	if reserveIpErr != nil {
-		t.Logf("Failed to reserve IP: %v", reserveIpErr)
-	} else {
-		t.Logf("Successfully reserved IP: %+v", reserveIP)
+
+	if resErr != nil {
+		t.Fatalf("Failed to reserve IP. This test expects the user to have 0 prior reservations and the ip_reservation_limit to be 2. Error from the API: %v", resErr)
 	}
 
-	if reserveIP != nil {
-		// Fetch the reserved IP
-		reservedIP, fetchIpErr := client.GetReservedIPAddress(context.Background(), reserveIP.Address)
-		if fetchIpErr != nil {
-			t.Logf("Error getting ipaddress, expected struct, got %v and error %v", reservedIP, fetchIpErr)
-		}
+	if resIP == nil {
+		t.Fatalf("Reserved IP is nil")
```

Comment:
Maybe state what the expected result in this case is, and why nil is a failure?

Reply:
Removed this condition as I dont believe there is a condition where the response would be nil without throwing an error. The error would be caught displayed to the user and test would consequently fail.